client/asset/interface.go

```diff
@@ -152,14 +153,24 @@ type Wallet interface {
 	// specified Coin. A slice of pubkeys required to spend the Coin and a
 	// signature for each pubkey are returned.
 	SignMessage(Coin, dex.Bytes) (pubkeys, sigs []dex.Bytes, err error)
-	// AuditContract retrieves information about a swap contract on the
-	// blockchain. This would be used to verify the counter-party's contract
-	// during a swap. If the coin cannot be found for the coin ID, the
-	// ExchangeWallet should return CoinNotFoundError. This enables the client
-	// to properly handle network latency. The matchTime is provided so that
-	// wallets can limit their scan when matching against transaction filters.
-	// necessary for wallets without full chain backing, but the caller should
-	// have it on hand anyway.
+	// AuditContract retrieves information about a swap contract from the
+	// blockchain (where possible) or from the provided txData (if valid).
+	// The information returned would be used to verify the counter-party's
+	// contract during a swap. If the coin cannot be found on the blockchain
+	// and the provided txData cannot be broadcasted, a CoinNotFoundError
+	// may be returned. This enables the client to properly handle network
+	// latency where appropriate.
+	//
+	// NOTE: For SPV wallets, a successful audit response is no gaurantee that
+	// the txData provided was actually broadcasted to the blockchain. An error
+	// may have occurred while trying to broadcast the txData or even if there
+	// was no broadcast error, the tx might still not enter mempool or get mined
+	// e.g. if the tx references invalid or already spent inputs.
+	//
+	// Granted, clients wait for the contract tx to be included in a block before
+	// taking further actions on a match; but it is generally safer to repeat this
+	// audit after the contract tx is mined to ensure that the tx observed on the
+	// blockchain is as expected.
```

Comment:
This is a red flag for me that says we should change the interface to separate audit of server-supplied transaction data from the confirmation of the tx on the blockchain. Both wallets can work fine with that model, and we wouldn't need notes like that basically just say that return values may mean different things depending on implementation-specific details.

Reply:
That can work, it's really just the same question of whether we want to audit the tx seen on the blockchain or just the txData alone. Confirming the tx on the blockchain doesn't audit it and we already have a separate method for that, so this could just simplify to auditing and broadcasting the txData. If we do want to audit the contract found on the blockchain, we could have `AuditContractData` and `AuditBlockchainContract` to simplify this method. Or do we just do away with repeat contract audits?